allennlp/tests/tools/drop_eval_test.py

```diff
@@ -139,3 +142,22 @@ def test_json_loader(self):
                                         {"answer": {"spans": ["answer2"]}, "query_id":"qid2"}]}}
         prediction = {"qid1": "answer", "qid2": "answer2"}
         assert evaluate_json(annotation, prediction) == (0.5, 0.5)
+
+    def test_type_partition_output(self):
+        annotation = {"pid1": {"qa_pairs":[{"answer": {"number": "5"}, "validated_answers": \
+                                                        [{"spans": ["7-meters"]}], "query_id":"qid1"}]}}
+        prediction = {"qid1": "5-yard"}
+        with io.StringIO() as buf, redirect_stdout(buf):
```

Comment:
I'm a bit confused about why you did this this way, instead of just doing `assert evaluate_json(annotation, prediction) == (...)`, as above.  Why doesn't that kind of test work?

Reply:
The output of `evaluate_json` is just `(global_em, global_f1)`. 
The change for `max_type` only affects the dictionaries `type_to_em` and `type_to_f1` which are only used in the printed output.

Reply:
Ok, great, thanks, I understand this better now.